Lib/ctypes/_aix.py

```
@@ -0,0 +1,265 @@
+https://www.ibm.com/support/knowledgecenter/en/ssw_aix_61/com.ibm.aix.basetrf1/load.htm
+"""
+__author__ = "Michael Felt <aixtools@felt.demon.nl>"
+__version__ = "1.0.0"
+
+# Latest Update (comments): 13 October 2016
+# Thanks to Martin Panter for his patience and comments
+
+from re import search, match, escape
+from os import environ, path
+from sys import executable
+from ctypes import c_void_p, sizeof
+from ctypes._util import _last_version
+from subprocess import Popen, PIPE
+
+def aixABI():
+    """Internal support function:
+    return executable size - 32-bit, or 64-bit
+    This is vital to the search for suitable member in an archive
+    """
+    return sizeof(c_void_p) * 8
+
+def get_dumpH(file):
+    """Internal support function:
+    dump -H output provides info on archive/executable contents
+    and related paths. This function call dump -H as a subprocess
+    and returns a list of (object, objectinfo) tuples.
+    """
+    #dumpH parsing:
+    # 1. When line starts with /, ./, or ../ - set as object
+    # 2. If "INDEX" in line object is confirmed as object
+    # 3. get info (lines starting with [0-9])
+
+    def get_object(p):
+        object = None
+        for line in p.stdout:
+            if line.startswith(('/', './', '../')):
+                object = line
+            elif "INDEX" in line:
+                return object.rstrip('\n')
+        return None
+
+    def get_info(p):
+        # as an object was found, return known paths, archives and members
+        # these lines start with a digit
+        info = []
+        for line in p.stdout:
+            if match("[0-9]", line):
+                info.append(line)
+            else:
+                # Should be a blank separator line, safe to consume
+                break
+        return info
+
+    p = Popen(["/usr/bin/dump", "-X%s" % aixABI(), "-H", file],
+        universal_newlines=True, stdout=PIPE)
```

You may redirect stderr to subprocess.DEVNULL, or to stdout (subprocess.STDOUT), to hide errors.

---

a) these options were chosen to be suitable for python2 as well as python3. Will have to read the documentation.
b) will redirect stderr to DEVNULL